src/entity/posts/utils.ts

```diff
@@ -352,6 +353,14 @@ interface CreateSharePostArgs {
   args: SharePostArgs;
 }
 
+export const determineSharedPostId = (post: Post | SharePost): string => {
+  if (post.type === PostType.Share && !post.title) {
+    return post instanceof SharePost ? post.sharedPostId : post.id;
```

Comment:
```suggestion
    if (post instanceof SharePost) {
        return post.sharedPostId
    }
```

this way we can add more conditions later below because it will fall-through to default `post.id`